mnd/mnd/views/mnd_form_view.py

```diff
@@ -0,0 +1,22 @@
+import logging
+
+from django.core.exceptions import PermissionDenied
+
+from rdrf.models.definition.models import RegistryForm
+from rdrf.security.security_checks import get_object_or_permission_denied
+from rdrf.views.form_view import FormView
+
+from registry.patients.models import Patient
+
+logger = logging.getLogger(__name__)
+
+
+class MNDFormView(FormView):
+    def registry_permissions_check(self, request, registry_code, form_id, patient_id, context_id):
+        if request.user.is_clinician:
+            form = get_object_or_permission_denied(RegistryForm, pk=form_id)
+            patient = get_object_or_permission_denied(Patient, pk=patient_id)
+
+            # TODO: Use upcoming patient-clinician connection feature
+            if not form.name.startswith("clinical") and patient.clinician != request.user:
```

Comment:
I was hoping to use `form.groups_allowed.filter(name__iexact=RDRF_GROUPS.PATIENT).exists()` instead of reading the form name, as that would be a _little_ less hacky.

But then we'd need remove the `Patient` group from clinical forms, which would allow working group clinicians to see them, but not patients...

The other way to do it would be adding a group called `Patients only` or something, which isn't really in the spirit of how groups should work